hvplot/tests/testibis.py

```diff
@@ -0,0 +1,73 @@
+"""Ibis works with hvplot"""
+import sqlite3
+from pathlib import Path
+
+import holoviews as hv
+import matplotlib
+import pytest
+
+matplotlib.use('agg')
+
+try:
+    import duckdb
+    import hvplot.ibis  # noqa
+    import hvplot.pandas  # noqa
+    import ibis
+    import pandas as pd
+except:
+    pytest.skip(allow_module_level=True)
+
+@pytest.fixture
+def reference_df():
+    return pd.DataFrame(
+        {
+            "actual": [100, 150, 125, 140, 145, 135, 123],
+            "forecast": [90, 160, 125, 150, 141, 141, 120],
+            "numerical": [1.1, 1.9, 3.2, 3.8, 4.3, 5.0, 5.5],
+            "date": pd.date_range("2022-01-03", "2022-01-09"),
+            "string": ["Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"],
+        },
+    )
+
+def pandas_data(df: pd.DataFrame, *args, **kwargs):
+    return df.copy()
+
+def ibis_sqlite_data(df: pd.DataFrame, *args, **kwargs):
+    tmpdir = kwargs["tmpdir"]
+    filename = str(Path(tmpdir)/"db.db")
+    con = sqlite3.Connection(filename)
+    df.to_sql("df", con, index=False)
+    con = ibis.sqlite.connect(filename)
+    return con.table("df")
+
+def ibis_pandas_data(df: pd.DataFrame, *args, **kwargs):
+    return ibis.pandas.connect({"df": df}).table("df")
+
+def ibis_duckdb_data(df: pd.DataFrame, *args, **kwargs):
+    tmpdir = kwargs["tmpdir"]
+    filename = str(Path(tmpdir)/"db.db")
+    duckdb_con = duckdb.connect(filename)
+    duckdb_con.execute("CREATE TABLE df AS SELECT * FROM df")
+
+    return ibis.duckdb.connect(filename).table("df")
+
+@pytest.fixture(params=[pandas_data, ibis_pandas_data, ibis_duckdb_data, ibis_sqlite_data])
+def data(request, reference_df, tmpdir):
+    return request.param(reference_df, tmpdir=tmpdir)
+
+@pytest.fixture(params=["bokeh", "matplotlib", "plotly"])
+def backend(request):
+    return request.param
+
+@pytest.fixture(params=["date", "numerical", "string"])
+def xseries(request):
+    return request.param
+
+def test_can_hvplot(xseries, data, backend):
```

Comment:
This is testing line plots, right?

Reply:
I can appreciate that end-to-end testing is useful but ideally we would continue to separate concerns, i.e.:

- HoloViews interface tests assert that the Ibis interface returns the data correctly
- HoloViews plotting tests assert that a correctly formed Element renders correctly (independent of the data backend)
- hvPlot asserts that it outputs correctly formed Elements

Reply:
I think so as well. But please note. I am close to a first time contributor. I have no experience in how this is organized and should work. I need help to see how this should be organized.

I just got rather scared when starting to work on this because hvplot 0.8.1 announces support for ibis and then the most simple example does not work and there is no tests. So I started adding at least some tests. I also needed to start from them since I did not know where the core error was.

Reply:
Just noting that Ibis support has been added in hvPlot [**0.7.1**](https://hvplot.holoviz.org/releases.html#version-0-7-1), not 0.8.1., and probably announced nowhere else. Doesn't mean it doesn't need to be well supported though!